osu.Game/Online/Metadata/OnlineMetadataClient.cs

```diff
@@ -242,7 +242,14 @@ public override async Task EndWatchingUserPresence()
                     throw new OperationCanceledException();
 
                 // must be scheduled before any remote calls to avoid mis-ordering.
-                Schedule(() => userStates.Clear());
+                Schedule(() =>
+                {
+                    bool hadLocalUserState = userStates.TryGetValue(api.LocalUser.Value.OnlineID, out var presence);
+                    userStates.Clear();
+                    if (hadLocalUserState)
+                        userStates[api.LocalUser.Value.OnlineID] = presence;
```

Comment:
As mentioned IRL, let's keep the local user state as a separate field to simplify this.